### PR TITLE
feat: debouncefield component

### PR DESCRIPTION
**Pull request by jackmellis**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,25 @@
...
+}) => (
+  <input
+    {...input}
+    {...rest}
+  />
+);
+
+const MyForm = () => (
+  <DebounceField
+    wait={250}
```
Comment (jshthornton) [inline README.md]: Does this have a default? If it does we should document the props to this component and what their default are.

File: package.json [hunk 2]
```diff
@@ -0,0 +1,108 @@
+{
+  "name": "redux-debounce-field",
```
Comment (jshthornton) [inline package.json]: Wouldn't this be better named `redux-form-debounce-field`, as that is what it is actually a plugin of.

File: package.json [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+  "version": "1.0.0",
+  "main": "dist/cjs/redux-debounce-field.js",
+  "module": "dist/es/redux-debounce-field.js",
+  "types": "dist/es/index.d.ts",
+  "repository": "git@github.secureserver.net:pexux/ux-redux-debounce-field",
+  "author": "Jack Ellis <jack.m.ellis@hotmail.co.uk>",
+  "license": "UNLICENCED",
+  "publishConfig": {
+    "registry": "https://artifactory.ba.heg.com/api/npm/godaddy-emea/",
```
Comment (jshthornton) [inline package.json]: Can you not only remove this but scrub the commit. Don't really want these urls in open source.

File: src/components/DebounceInput.tsx [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+  ...props
+}: InnerProps) => (
+  <Component {...props}/>
+);
+
+const enhance = compose<InnerProps, OuterProps>(
+  setDisplayName('DebounceInput'),
+  pure,
+  defaultProps({ wait: 250 }),
+  withState('debounceFieldValue', 'setDebounceFieldValue', ''),
```
Comment (jshthornton) [inline src/components/DebounceInput.tsx]: If you have multiple `withState` hocs think of using: https://github.com/acdlite/recompose/blob/master/docs/API.md#withstatehandlers

File: README.md (hunk 1 above)
Comment (jackmellis) [reply README.md]: yes it does have a default, I've updated the docs

File: package.json (hunk 2 above)
Comment (jackmellis) [reply package.json]: renamed, we also rename the repo

File: src/components/DebounceInput.tsx (hunk 4 above)
Comment (jackmellis) [reply src/components/DebounceInput.tsx]: done

**Pull request merged**